src/main/java/codechicken/nei/recipe/RecipeItemInputHandler.java

```diff
@@ -39,68 +42,84 @@ public boolean lastKeyTyped(GuiContainer gui, char keyChar, int keyCode)
             ItemPanels.bookmarkPanel.addOrRemoveItem(stackover.copy(), handlerName, ingredients);
         }
 
+        if (keyCode == NEIClientConfig.getKeyBinding("gui.overlay")) {
+            final Point mousePosition = GuiDraw.getMousePosition();
+            if (ItemPanels.bookmarkPanel.getStackMouseOver(mousePosition.x, mousePosition.y) == null) return false;
+            final BookmarkRecipeId recipeId = ItemPanels.bookmarkPanel.getBookmarkRecipeId(stackover);
+            if (recipeId == null || recipeId.handlerName == null || recipeId.ingredients == null || recipeId.ingredients.isEmpty())
+                return false;
+            final ArrayList<ICraftingHandler> handlers = GuiCraftingRecipe.getHandlers("item", recipeId.handlerName, stackover.copy());
+            if (handlers == null || handlers.isEmpty()) return false;
+            final ICraftingHandler handler = handlers.get(0);
+            if (recipeId.position == -1) {
+                for (int i = 0; i < handler.numRecipes(); i++) {
+                    if (recipeId.equalsIngredients(handler.getIngredientStacks(i))) {
+                        recipeId.position = i;
+                        break;
+                    }
+                }
+            }
+            if (recipeId.position == -1) return false;
+            final IRecipeOverlayRenderer renderer = handler.getOverlayRenderer(gui, recipeId.position);
+            final IOverlayHandler overlayHandler = handler.getOverlayHandler(gui, recipeId.position);
+            final boolean shift = NEIClientUtils.shiftKey();
+            if (renderer == null || shift) {
+                overlayHandler.overlayRecipe(gui, handler, recipeId.position, shift);
+            } else {
+                LayoutManager.overlayRenderer = LayoutManager.overlayRenderer == null ? renderer : null;
```

Comment:
I think overlaying another recipe should be rendered, rather than just removing current overlay

Reply:
ok